src/cc/bcc_proc.c

```diff
@@ -148,39 +143,66 @@ int bcc_procutils_each_module(int pid, bcc_procutils_modulecb callback,
     name = buf;
     while (isspace(*name))
       name++;
+    mod.name = name;
     if (!bcc_mapping_is_file_backed(name))
       continue;
 
     if (strstr(name, "/memfd:")) {
-      char *memfd_name = _procutils_memfd_path(pid, inode);
+      char *memfd_name = _procutils_memfd_path(pid, mod.inode);
       if (memfd_name != NULL) {
         strcpy(buf, memfd_name);
         free(memfd_name);
-        name = buf;
+        mod.name = buf;
+        enter_ns = 0;
       }
     }
 
-    if (callback(name, begin, end, (uint64_t)offset, true, payload) < 0)
-      break;
+    if (callback(&mod, enter_ns, payload) < 0)
+      return 0;
   }
 
-  fclose(procmap);
+  return -1;
+}
+
+int bcc_procutils_each_module(int pid, bcc_procutils_modulecb callback,
+                              void *payload) {
+  char procmap_filename[128];
+  FILE *procmap;
+  snprintf(procmap_filename, sizeof(procmap_filename), "/proc/%ld/maps",
+           (long)pid);
+  procmap = fopen(procmap_filename, "r");
+  if (!procmap)
+    return -1;
+
+  if(!_procfs_maps_each_module(procmap, pid, callback, payload))
+    goto done;
```

Comment:
this is different from original implementation. Previously, regardless return value, the below perf-<PID>.map is searched. This is not the case any more?

Reply:
this was a mistake on my part, changed to function as before